internal/guest/storage/scsi/scsi.go

```diff
@@ -245,8 +239,12 @@ func unmount(
 	}
 
 	if encrypted {
-		if err := crypt.CleanupCryptDevice(target); err != nil {
-			return errors.Wrapf(err, "failed to cleanup dm-crypt state: "+target)
+		source, err := controllerLunToName(ctx, controller, lun)
```

Comment:
refactored code to pass the dm-crypt name rather than calling to `getUniqueName`. As a result, updated the unit tests and removed unnecessary functions and constants.